catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -96,6 +99,14 @@ public class HadoopCatalogOperations implements CatalogOperations, SupportsSchem
 
   private CatalogInfo catalogInfo;
 
+  private final List<Closeable> closeables = Lists.newArrayList();
+
+  private final Map<NameIdentifier, UserInfo> userInfoMap = Maps.newConcurrentMap();
+
+  public static final String GRAVITINO_KEYTAB_FORMAT = "keytabs/gravitino-%s";
+
+  private final ThreadLocal<String> currentUser = ThreadLocal.withInitial(() -> null);
```

Comment:
Could you avoid using thread local variable here? We can doAs in the wrapper like other code.

Reply:
I used a local thread to keep the API user or we won't be able to get the API user in the `doAs` block.